### PR TITLE
cifsd: fix some smb2.setinfo failures

### PR DESCRIPTION
**Pull request by tymok**
1. make setinfo not to allow to change non-directory file's attributes
into ATTR_DIRECTORY.

2. implement set_info FILE_POSITION_INFORMATION case.

Signed-off-by: Taeyang Mok <t.mok@samsung.com>

### CODE REVIEW

File: misc.c [hunk 1]
```diff
@@ -892,3 +894,23 @@ int construct_xattr_stream_name(char *stream_name, char **xattr_stream_name)
...
 	return XATTR_NAME_STREAM_LEN + stream_size;
 }
+
+/*
+ * get_logical_sector_size() - get logical sector size from inode
+ * @inode: inode
+ *
+ * Return: logical sector size
+ */
+unsigned short get_logical_sector_size(struct inode *inode)
```
Comment (namjaejeon) [inline misc.c]: I thnk that this function is added in vfs.c file.

File: smb2pdu.c [hunk 2]
```diff
@@ -5061,6 +5062,14 @@ int smb2_set_info_file(struct smb_work *smb_work)
 		if (le32_to_cpu(file_info->Attributes)) {
 			unsigned long *config_attr;
 
+			if (!S_ISDIR(file_inode(filp)->i_mode)
+				&& file_info->Attributes == ATTR_DIRECTORY) {
+				cifsd_debug("can't change a file to a directory\n");
```
Comment (namjaejeon) [inline smb2pdu.c]: if this is error that not happen redundantly, need to show error message using cifsd_err.

File: smb2pdu.c [hunk 3]
```diff
@@ -5061,6 +5062,14 @@ int smb2_set_info_file(struct smb_work *smb_work)
...
+				rsp->hdr.Status = NT_STATUS_INVALID_PARAMETER;
+				smb2_set_err_rsp(smb_work);
+				return -EPERM;
```
Comment (namjaejeon) [inline smb2pdu.c]: if you use -ENVIAL return, no need to set status and use smb2_set_err_rsp.

File: smb2pdu.c [hunk 4]
```diff
@@ -5223,6 +5232,29 @@ int smb2_set_info_file(struct smb_work *smb_work)
...
+			current_byte_offset & (sector_size-1))) {
+			cifsd_err("CurrentByteOffset is not valid : %llu\n",
+				current_byte_offset);
+			rsp->hdr.Status = NT_STATUS_INVALID_PARAMETER;
+			return -EINVAL;
+		}
+
+		filp->f_pos = current_byte_offset;
+
+		break;
```
Comment (namjaejeon) [inline smb2pdu.c]: looks good!

**Comment by tymok** [comment]
Thanks for the review. I modified the error handlings and moved get_logical_sector_size into vfs.c

**Comment by tymok** [comment]
As we discussed, I am going to merge all the patch to one commit and commit it again.

**Comment by tymok** [comment]
A new patch merged with https://github.com/namjaejeon/cifsd/pull/16

**Pull request closed without merging**